indexing-service/src/main/java/org/apache/druid/indexing/common/index/YeOldePlumberSchool.java

```diff
@@ -113,7 +113,7 @@ public Plumber findPlumber(
     );
 
     // Temporary directory to hold spilled segments.
-    final File persistDir = new File(tmpSegmentDir, theSink.getSegment().getIdentifier());
+    final File persistDir = new File(tmpSegmentDir, theSink.getSegment().getId().toString());
```

Comment:
This is an unfortunate legacy effect. It would be nice if we didn't inject segment ids as file names. That can cause odd problems depending on the file system. Probably out of scope for this PR though

Reply:
Could you please create an issue